_posts/2021-05-10-helm-charts-for-wildfly.adoc

```diff
@@ -0,0 +1,225 @@
+
+The `s2i` build mode uses WildFly Source-to-Image (S2I) Builder and Runtime images to create the application image.  
+In the `bootable-jar` build mode, the `BuildConfig` will compile the application as a Bootable Jar and use the OpenJDK 11 image as the base image.
+
+Finally there is an `env` section that contains any environment variables needed to *build* the image. We have three environment variables that are needed for Maven options.
+As a whole they mean that we build only the specific `microprofile-config` Maven module from the quickstart Git repository with the `bootable-jar-openshift` Maven Profile (to create a Bootable Jar) and use its artifact as the target of the application image.
+
+The `deploy` section focuses on deploying the application image on OpenShift. It creates different resources (`Deployment`, one or many `Services`, a `Route`) to make the application accessible from inside and outside the cluster by default.
+
+We have only defined two fields in that section:
+
+* `replicas` which specifies the number of pods that the application will use
+* `env` wich are environment variables needed to *run* the image. In our example, we only have one named `CONFIG_PROP` and its value is used in the HTTP endpoint's response.
+
+There are a lot of things to discuss with this chart and we will have other blog post to showcase its features.
+All of them are documented in the https://github.com/wildfly/wildfly-charts/blob/main/charts/wildfly/README.md[`wildfly` Chart documentation].
+
+However as a last example, one of the interesting feature of Helm is that the YAML configuration file can be overridden on the command line to provide additional customization to the application.
+
+To highlight this, we will *upgrade* the application by changing the number of replicas of the applications so that we have `3` pods instead of `1` (by setting the `deploy.replicas` field to 3 with the `helm upgrade` command)
+
+[source,nowrap]
+----
+$ helm upgrade microprofile-config-app \
+    wildfly/wildfly \
+    -f https://raw.githubusercontent.com/wildfly/wildfly-charts/main/examples/microprofile-config/microprofile-config-app.yaml \
+    --set deploy.replicas=3 
+Release "microprofile-config-app" has been upgraded. Happy Helming!
+NAME: microprofile-config-app
+LAST DEPLOYED: Tue May  4 15:21:29 2021
+NAMESPACE: jmesnil1-dev
+STATUS: deployed
+REVISION: 2
+TEST SUITE: None
+NOTES:
+----
+
+If we watch the deployment, we see that the application is now scaling up to 3:
+
+[source]
+----
+$ oc get deployment microprofile-config-app -w
+NAME                      READY   UP-TO-DATE   AVAILABLE   AGE
+microprofile-config-app   1/3     3            1           29m
+microprofile-config-app   2/3     3            2           30m
+microprofile-config-app   3/3     3            3           30m
+----
+
+## Conclusion
+
+The Helm Charts for WildFly simplifies building and deploying WildFly application on OpenShift and Kubernetes.
```

Comment:
s/Charts/Chart  or s/simplifies/simplify

Then in the next sentence you use "They", implying more than 1.

There is more than one but maybe here just use "Chart" and "It" instead of "They"?  This post doesn't really get into the chart architecture, so the plural here is correct but confusing.